File: September 2026/2026-09-11/3483-unique-3digit-even-number.cpp
// 3483-unique-3digit-even-number;array;self;leetcode

class Solution {
public:
    int totalNumbers(vector<int>& digits) {
        set<int> nums;
        int n = digits.size();
        for(int i = 0; i < n; i++) {
            if(digits[i] == 0) {
                continue;
            }

            for(int j = 0; j < n; j++) {
                if(j == i) continue;

                for(int k = 0; k < n; k++) {
                    if (k == i || k == j)
                        continue;
                    if (digits[k] % 2 != 0)
                        continue;

                    int number = digits[i] * 100 + digits[j] * 10 + digits[k];
                    nums.insert(number);
                }
            }
        }
        return nums.size();
    }
};

File: September 2026/2026-09-11/3483-unique-3digit-even-number.py
# 3483-unique-3digit-even-number;array;self;leetcode

class Solution:
    def totalNumbers(self, digits: List[int]) -> int:
        nums = set()

        for i in range(len(digits)):

            if digits[i] == 0:
                continue

            for j in range(len(digits)):

                if j == i:
                    continue

                for k in range(len(digits)):
                    if k == i or k == j:
                        continue
                    if digits[k] % 2 != 0:
                        continue
                    number = digits[i] * 100 + digits[j] * 10 + digits[k]

                    nums.add(number)
        return len(nums)